feat: Allow configuration of ExternalAuthorizer in IstioCR (#639)
kyma-project/istio: docs/user/04-00-istio-custom-resource.md

| **components.proxy.k8s.resources**                    | object         | Defines [Kubernetes resources requests and limits configuration](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/). To learn more, read about Resources in the [Istio documnetation](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Resources).                                       |
| **config**                                            | object         | Specifies the configuration for the Istio installation.                                                                                                                                                                                                                                                                                     |

> | **config.authorizers**                                | \[\]authorizer | Specifies the list of external authorizers that will be configured in Istio mesh config.                                                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| **config.authorizers**                                | \[\]authorizer | Specifies the list of external authorizers configured in the Istio service mesh config.                                                                                                                                                                                                                                                    |
```

| **config.numTrustedProxies**                          | integer        | Specifies the number of trusted proxies deployed in front of the Istio gateway proxy.                                                                                                                                                                                                                                                       |

**Authorizer**

| Parameter              | Type    | Description                                                                                                                                 |
|------------------------|---------|---------------------------------------------------------------------------------------------------------------------------------------------|
| **name** (required)    | string  | A unique name identifying the extension authorization provider.                                                                             |

> | **service** (required) | string  | Specifies the service that implements the Envoy ext_authz HTTP authorization service. The recommended format is "[<Namespace>/]<Hostname>". |

Reviewer: ```suggestion
| **service** (required) | string  | Specifies the service that implements the Envoy `ext_authz` HTTP authorization service. The recommended format is `[<Namespace>/]<Hostname>`. |
```

> | **port** (required)    | integer | Specifies the port of the external authorizer on which the authorization request is made.                                                   |

Reviewer: ```suggestion
| **port** (required)    | integer | Specifies the port number of the external authorizer used to make the authorization request.                                                  |
```

> | **headers**            | headers | Specifies headers to be included, added or forwarded during authorization.                                                                  |

Reviewer: ```suggestion
| **headers**            | headers | Specifies headers to be included, added, or forwarded during authorization.                                                                  |
```

**Headers**

| Parameter                | Type       | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
|--------------------------|------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | **inCheck**              | object     | Defines headers to be included or added in check authorization request.                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| **inCheck**              | object     | Defines headers to be included or added in the authorization request check.                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
```

> | **inCheck.include**      | \[\]string | List of client request headers that should be included in the authorization request sent to the authorization service.                                                                                                                                                                                                                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| **inCheck.include**      | \[\]string | Specifies a list of client request headers that should be included in the authorization request sent to the authorization service.                                                                                                                                                                                                                                                                                                                                                                                                                    |
```

> | **inCheck.add**          | key-value  | Set of additional fixed headers that should be included in the authorization request sent to the authorization service. The Key is the header name and value is the header value. Note that client request of the same key or headers specified in `Include` will be overridden.                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| **inCheck.add**          | key-value  | Defines a set of additional fixed headers that should be included in the authorization request sent to the authorization service. The key is the header's name, and the value is the header's value. It overrides a client request of the same key or headers specified in the **include** field.                                                                                                                                                                                                                                                          |
```

| **toUpstream**           | object     | Defines headers to be forwarded to the upstream.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |

> | **toUpstream.onAllow**   | \[\]string | List of headers from the authorization service that should be added or overridden in the original request and forwarded to the upstream when the authorization check result is allowed (HTTP code 200). If not specified, the original request will not be modified and forwarded to backend as-is. Note, any existing headers will be overridden.                                                                                                                                                                                        | 

Reviewer: ```suggestion
| **toUpstream.onAllow**   | \[\]string | Specifies a list of headers from the authorization service that should be added or overridden in the original request and forwarded to the upstream when the authorization check result is allowed (HTTP code `200`). If not specified, the original request will not be modified and forwarded to the backend as-is. It overrides any existing headers.                                                                                                                                                                                        | 
```

| **toDownstream**         | object     | Defines headers to be forwarded to the downstream.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |

> | **toDownstream.onAllow** | \[\]string | List of headers from the authorization service that should be added or overridden in the original request and forwarded to the upstream when the authorization check result is allowed (HTTP code 200). If not specified, the original request will not be modified and forwarded to backend as-is. Note, any existing headers will be overridden.                                                                                                                                                                                        | 

Reviewer: ```suggestion
| **toDownstream.onAllow** | \[\]string | Specifies a list of headers from the authorization service that should be added or overridden in the original request and forwarded to the upstream when the authorization check result is allowed (HTTP code `200`). If not specified, the original request will not be modified and forwarded to backend as-is. It overrides any existing headers.                                                                                                                                                                                   | 
```

> | **toDownstream.onDeny**  | \[\]string | List of headers from the authorization service that should be forwarded to downstream when the authorization check result is not allowed (HTTP code other than 200). If not specified, all the authorization response headers, except *Authority (Host)* will be in the response to the downstream. When a header is included in this list, *Path*, *Status*, *Content-Length*, *WWWAuthenticate* and *Location* are automatically added. Note, the body from the authorization service is always included in the response to downstream. | 

Reviewer: ```suggestion
| **toDownstream.onDeny**  | \[\]string | List of headers from the authorization service that should be forwarded to downstream when the authorization check result is not allowed (HTTP code other than `200`). If not specified, the response to the downstream will contain all the authorization response headers, except **Authority (Host)**. When a header is included in this list, **Path**, **Status**, **Content-Length**, **WWWAuthenticate**, and *Location* are automatically added. Note that the body from the authorization service is always included in the response to downstream.  | 
```
